src/app/system/manage-data-tables/view-data-table/view-data-table.component.ts

```diff
@@ -0,0 +1,74 @@
+/** Angular Imports */
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { MatTableDataSource, MatPaginator, MatSort } from '@angular/material';
+
+/** Custom Services */
+import { SystemService } from '../../system.service';
+
+@Component({
+  selector: 'mifosx-view-data-table',
+  templateUrl: './view-data-table.component.html',
+  styleUrls: ['./view-data-table.component.scss']
+})
+
+/**
+ * View Data Table Component
+ */
+export class ViewDataTableComponent implements OnInit {
+
+  /** Data Table Data */
+  dataTableData: any;
+  /** Column Data */
+  columnsData: any;
+  /** Columns to be displayed in columns table. */
+  displayedColumns: string[] = ['columnName', 'columnType', 'columnLength', 'columnCode', 'isColumnPrimaryKey'];
+  /** Data source for columns table. */
+  dataSource: MatTableDataSource<any>;
+
+  /** Paginator for columns table. */
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  /** Sorter for columns table. */
+  @ViewChild(MatSort) sort: MatSort;
+
+  /**
+   * Retrieves the data table data from `resolve`.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {SystemService} systemService System Service.
+   */
+  constructor(private route: ActivatedRoute,
+              private systemService: SystemService,
+              private router: Router) {
+    this.route.data.subscribe((data: { dataTable: any }) => {
+      this.dataTableData = data.dataTable;
+      this.columnsData = this.dataTableData.columnHeaderData;
+    });
+  }
+
+  /**
+   * Sets the columns table.
+   */
+  ngOnInit() {
+    this.setColumnsTable();
+  }
+
+  /**
+   * Initializes the data source, paginator and sorter for columns table.
+   */
+  setColumnsTable() {
+    this.columnsData.shift();
+    this.dataSource = new MatTableDataSource(this.columnsData);
+    this.dataSource.paginator = this.paginator;
+    this.dataSource.sort = this.sort;
+  }
+
+  /**
+   * Deletes the current data table.
+   */
+  delete() {
+    this.systemService.deleteDataTable(this.dataTableData.registeredTableName)
```

Comment:
Should open the `delete-dialog` and ask for confirmation before deletion. See accounting section for reference.